### PR TITLE
Story/vspc 21

### PR DESCRIPTION
**Pull request by Neehaarika**
Had to recreate this branch as there was a diverge issue with the old one.

### CODE REVIEW

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/module.jsp [hunk 1]
```diff
@@ -1,66 +1,114 @@
...
 	$("#cancelSlideBtn").click(function() {
-		$("#createSlideAlert").hide();	
+		$("#createSlideAlert").hide();
 	});
-});
-
+	
+	$(".sequence").on("click", function(e) {
+		$(".sequence").css({ 'border' : ''});
+		$(this).css("border", "solid deepSkyBlue");
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/module.jsp]: indentation

File: vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SequenceManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    @Test
+    public void test_getSequence_idExists() {
+        Sequence newSequence = new Sequence();
+        newSequence.setId("sequence1");
+        Optional<Sequence> sequence = Optional.of(newSequence);
+        Mockito.when(mockSequenceRepo.findById(newSequence.getId())).thenReturn(sequence);
+
+        ISequence actualSequence = sequenceManagerToTest.getSequence(newSequence.getId());
+        Assert.assertEquals(newSequence.getId(), actualSequence.getId());
+        Assert.assertEquals(newSequence.getName(), actualSequence.getName());
```
Comment (jdamerow) [inline vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SequenceManagerTest.java]: you never set the name of the sequence

File: vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SequenceManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        newSequence.setId("seq01");
+        newSequence.setModule(module);
+        newSequence.setName(sequenceForm.getName());
+        newSequence.setDescription(sequenceForm.getDescription());
+        newSequence.setSlides(slidesList);
+              
+        Mockito.when(slideManager.getSlide("slide1")).thenReturn(slidesList.get(0));
+        Mockito.when(slideManager.getSlide("slide2")).thenReturn(slidesList.get(1));
+        Mockito.when(moduleManager.getModule("module1")).thenReturn(module);
+        Mockito.when(sequenceFactory.createSequence(module, sequenceForm, slidesList)).thenReturn(newSequence);
```
Comment (jdamerow) [inline vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SequenceManagerTest.java]: you want to make sure that the factory returns a sequence object without an id. the sequence gets an id from the repository.

**Comment by jdamerow** [comment]
I believe in the mock up, the slides in a sequence were supposed to be shown on the module page:
https://diging.atlassian.net/wiki/spaces/VSPACE/pages/1068826625/Layout+mockups+and+stuff
(not on a new page). Since there is just one start sequence, it doesn't make sure to have a whole column for that (it should just be a property on the page). I would show all slides in a sequence in the third column and keep a "details" link on each sequence card that opens the new page with all details (we might add more details later).

**Comment by Neehaarika** [comment]
After debugging I could see the new Sequence slides upon selection are being displayed in the module page itself. I have added details icon which will navigate to new Sequence page upon selecting. Please review the changes.

**Comment by jdamerow** [comment]
Jenkins, test this please.

File: vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SequenceManagerTest.java (hunk 2 above)
Comment (jdamerow) [reply vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SequenceManagerTest.java]: ?

File: vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SequenceManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        
+        ISequence actualSequence = sequenceManagerToTest.storeSequence(module.getId(), sequenceForm);
+
+        Assert.assertEquals(actualSequence.getModule().getId(), module.getId());
+        Assert.assertEquals(actualSequence.getName(), sequenceForm.getName());
+        Assert.assertEquals(actualSequence.getDescription(), sequenceForm.getDescription());
+        Assert.assertEquals(actualSequence.getSlides(), slidesList);
+        Mockito.verify(mockSequenceRepo).save(newSequence); 
+        Assert.assertNotNull(actualSequence.getId());
+    }
```
Comment (jdamerow) [inline vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SequenceManagerTest.java]: I think I've alerady commented on this in a different branch?

**Comment by jdamerow** [comment]
The cards seem to go further and further to the left so that are not all aligned with the column header:
image
Also, Column header of right most column says "Start Sequence" but should say something like "Slides in Sequence". And maybe choose a highlight color for the selected sequence that goes with the rest of the layout ;)

**Comment by jdamerow** [comment]
please resolve conflicts

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/module.jsp [hunk 5]
```diff
@@ -5,16 +5,42 @@
...
+               	  $('#selectedSequence').append(''+
+               		'<div class="card sequence" style="max-width: 18rem; margin-bottom:10px;">'+
+               		'<div align="left" class="card-body">'+
+               		'<a href="<c:url value="/staff/module/${module.id}/slide/"/>'+slide.id+'" >'+
+               		'<h5 class="card-title">'+slide.name+'</h5>'+
+               		'<p class="card-text">'+slide.description+'</p>'+'</a>'+
+               		'</div></div>');
+               	  });
+               	}
+            });
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/module.jsp]: indentation

**Comment by jdamerow** [comment]
nothing happens when I click on a sequence (no slides are shown in the right column, only the sequence is highlighted) and the third column is still title "Start Sequence".

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Pull request merged**